pgmpy/models/LinearGaussianBayesianNetwork.py

```diff
@@ -272,18 +290,126 @@ def simulate(self, n_samples=1000, seed=None):
+            for var, val in do.items():
+                for parent in list(model.get_parents(var)):
+                    model.remove_edge(parent, var)
+
+                    new_cpd = LinearGaussianCPD(
+                        variable=var, beta=[val], std=1e-3, evidence=[]
+                    )
+                    model.remove_cpds(model.get_cpds(var))
+                    model.add_cpds(new_cpd)
+
+        else:
+            model = self
+
+        if missing_prob is not None:
+            if isinstance(missing_prob, list):
+                for cpd in missing_prob:
+                    if not isinstance(cpd, LinearGaussianCPD):
+                        raise ValueError(
+                            f"missing_prob must be a list of LinearGaussianCPD objects. Got {type(cpd)}"
+                        )
+            else:
+                if isinstance(missing_prob, LinearGaussianCPD):
+                    missing_prob = [missing_prob]
+                else:
+                    raise ValueError(
+                        f"missing_prob should be LinearGaussianCPD. Got {type(missing_prob)}"
+                    )
+
+            for cpd in missing_prob:
+                missing_var = cpd.variables[0]
+
+                if not missing_var.endswith("*"):
+                    raise ValueError(
+                        f"Got variable '{missing_var}'. Missingness variable should end with '*' to represent missingness (e.g., 'X*')."
+                    )
+
+                base_var = missing_var[:-1]
+
+                if base_var not in model.nodes:
+                    raise ValueError(
+                        f"Missingness variable '{missing_var}' refers to base variable '{base_var}', which is not in model nodes."
+                    )
+
+                model.add_node(missing_var)
+
+                if len(cpd.variables) > 1:
+                    evidences = cpd.variables[1:]
+                    for parent in evidences:
+                        if parent not in model.nodes:
+                            raise ValueError(
+                                f"Missingness CPD refers to evidence variable '{parent}', which is not in the model."
+                            )
+                        model.add_edge(parent, missing_var)
+
+                model.add_cpds(cpd)
+
+        mean, cov = model.to_joint_gaussian()
+        variables = list(nx.topological_sort(model))
         rng = np.random.default_rng(seed=seed)
-        return pd.DataFrame(
-            rng.multivariate_normal(mean=mean, cov=cov, size=n_samples),
-            columns=variables,
-        )
+
+        evidence_var = list(evidence.keys())
+        sample_var = [v for v in variables if v not in evidence_var]
+
+        if len(evidence) == 0:
+            df = pd.DataFrame(
+                rng.multivariate_normal(mean=mean, cov=cov, size=n_samples),
+                columns=variables,
+            )
+
+        else:
+            df = pd.DataFrame([evidence])
```

Comment:
You will need to compute the joint distribution and then apply the evidence to compute the posterior joint distribution.